Assets/Sun.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Sun : MonoBehaviour
+{
+    [SerializeField, Header("Sun")]
+    public float rotationSpeed;
+
+    [SerializeField]
+    public float time;
+
+    [SerializeField]
+    public AnimationCurve lightIntensity;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        GetComponent<Rigidbody>().angularVelocity = new Vector3(rotationSpeed, 0, 0);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        Light sunLight = GetComponent<Light>();

```

Comment:
Is a Light a monoidal datatype? If so we coudl instantiate a sunLight with a neutral value then foldLeft[MonoidalLight] to produce a Monoid[Light]